src/util/msvc.h

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#ifndef UTIL_MSVC_COMPILER_H
+#define UTIL_MSVC_COMPILER_H
+
+#ifdef _MSC_VER
+#if __cplusplus >= 201700L  // with C++17 it's this deprecated
+                            // struct not available anymore
+#   include "util/xcs_functional.hpp"
+#endif
+#   include <io.h>
+#   include "corecrt_math_defines.h"
+
+#define STRING2(x) #x
+#define STRING(x) STRING2(x)
+
+#define CLANG_OR_GCC_VERSION(x, y)  0
+#define GCC_CHECK_VERSION(x, y)     0
+#define GCC_OLDER_THAN(x, y)        1  // This isn't a (new!) GCC
```

Comment:
Wrong. Your implementation says "this is a GCC older than the version specified", but it's not a GCC at all, thus cannot be a GCC older than that.

Reply:
So  I have to use at every time where this is ask:
```
#if GCC_OLDER_THAN(x, y)  || defined(_MSC_VER)
```
Is this ok so ? Or did you have a better idea?

Reply:
Why would you want to do that?